SQL assignment.txt

```diff
@@ -0,0 +1,1533 @@
+1) List all the columns of the Salespeople table. 
+
+mysql> DESC SALESPEOPLE;
+
++-------+---------+------+-----+---------+----------------+
+| Field | Type    | Null | Key | Default | Extra          |
++-------+---------+------+-----+---------+----------------+
+| snum  | int     | NO   | PRI | NULL    | auto_increment |
+| sname | char(8) | NO   |     | NULL    |                |
+| city  | char(9) | NO   |     | NULL    |                |
+| comm  | int     | YES  | UNI | NULL    |                |
++-------+---------+------+-----+---------+----------------+
+
+
+
+2. List all customers with a rating of 100.
+
+mysql> SELECT * FROM CUSTOMERS WHERE RATING=100;
+
++------+----------+--------+--------+------+
+| cnum | cname    | city   | rating | snum |
++------+----------+--------+--------+------+
+| 2001 | hoffman  | london |    100 | 1001 |
+| 2006 | cisneras | london |    100 | 1001 |
+| 2007 | pereira  | rome   |    100 | 1004 |
++------+----------+--------+--------+------+
+
+
+
+3). Find the largest order taken by each salesperson on each date.
+
+mysql>  SELECT o.odate, o.snum, s.sname, MAX(o.amt) AS largest_order
+    -> FROM orders AS o
+    -> JOIN salespeople AS s ON o.snum = s.snum
+    -> GROUP BY o.odate, o.snum, s.sname;
+
++----------+------+---------+---------------+
+| odate    | snum | sname   | largest_order |
++----------+------+---------+---------------+
+| 10/03/90 | 1007 | rifkin  |       1098.16 |
+| 10/03/90 | 1001 | Peel    |        767.19 |
+| 10/03/90 | 1004 | motika  |        1900.1 |
+| 10/03/90 | 1002 | serres  |       5160.45 |
+| 10/04/90 | 1003 | axeirod |       1713.23 |
+| 10/04/90 | 1002 | serres  |         75.75 |
+| 10/05/90 | 1001 | Peel    |          4723 |
+| 10/06/90 | 1002 | serres  |       1309.95 |
+| 10/06/90 | 1001 | Peel    |       9891.88 |
++----------+------+---------+---------------+
+
+
+
+4) Arrange the Orders table by descending customer number.
+
+mysql> SELECT * FROM orders ORDER BY cnum;
+
++------+---------+----------+------+------+
+| onum | amt     | odate    | cnum | snum |
++------+---------+----------+------+------+
+| 3003 |  767.19 | 10/03/90 | 2001 | 1001 |
+| 3009 | 1713.23 | 10/04/90 | 2002 | 1003 |
+| 3005 | 5160.45 | 10/03/90 | 2003 | 1002 |
+| 3007 |   75.75 | 10/04/90 | 2004 | 1002 |
+| 3010 | 1309.95 | 10/06/90 | 2004 | 1002 |
+| 3008 |    4723 | 10/05/90 | 2006 | 1001 |
+| 3011 | 9891.88 | 10/06/90 | 2006 | 1001 |
+| 3002 |  1900.1 | 10/03/90 | 2007 | 1004 |
+| 3001 |   18.69 | 10/03/90 | 2008 | 1007 |
+| 3006 | 1098.16 | 10/03/90 | 2008 | 1007 |
++------+---------+----------+------+------+
+
+
+
+5) Find which salespeople currently have orders in the order table. 
```

Comment:
Use can also use Join to solve this query. Joins are efficient than Subquery.

Reply:
resolved